### PR TITLE
add tests for variable expansion

### PR DESCRIPTION
**Pull request by bparees**
fixes https://github.com/openshift/origin/pull/5903

### CODE REVIEW

**Comment by bparees** [comment]
[testonlyextended][extended:cmd]

**Comment by bparees** [comment]
@pmorie this was your ask, so ptal.

**Comment by pmorie** [comment]
Looks good so far, but add a test that composes one env w/ something from the downward API like:

```
POD_NAME=<pod name via downward API>
POD_NAME_COMPOSED=$(POD_NAME)_foo
```

**Comment by bparees** [comment]
@pmorie added.

File: test/extended/cmd.sh [hunk 1]
```diff
@@ -68,3 +68,14 @@ rm -rf $tmp
 [ "$(oc new-app test/scratchimage2 -o yaml |& tr '\n' ' ' 2>&1 | grep -E "partial match")" ]
 # success with exact match	
 [ "$(oc new-app test/scratchimage -o yaml)" ]
+
+echo "[INFO] Running env variable expansion tests"
+oc new-project envtest
+oc create -f test/extended/fixtures/test-env-pod.json
+tryuntil "oc get pods | grep Running"
+podname=$(oc get pods --template='{{with index .items 0}}{{.metadata.name}}{{end}}')
+oc exec test-pod env | grep containername=test-pod
```
Comment (pmorie) [inline test/extended/cmd.sh]: containername? seems like this should be podname from the json

Comment (bparees) [inline test/extended/cmd.sh]: indeed.  turns out this script wasn't properly exiting w/ failure.  fixed the name and the exit issue.

**Comment by pmorie** [comment]
LGTM [merge]

**Pull request merged**